src/hex.ts

```diff
@@ -55,6 +65,55 @@ export function assertIsStrictHexString(value: unknown): asserts value is Hex {
   );
 }
 
+/**
+ * Validate that the passed prefixed hex string is an all-lowercase
+ * hex address, or a valid mixed-case checksum address.
+ *
+ * @param possibleAddress - Input parameter to check against.
+ * @returns Whether or not the input is a valid hex address.
+ */
+export function isValidHexAddress(possibleAddress: Hex) {
+  return (
+    is(possibleAddress, HexAddressStruct) ||
+    isValidChecksumAddress(possibleAddress)
+  );
+}
+
+/**
+ * Encode a passed hex string as an ERC-55 mixed-case checksum address.
+ *
+ * @param address - The hex address to encode.
+ * @returns The address encoded according to ERC-55.
+ * @see https://eips.ethereum.org/EIPS/eip-55
+ */
+export function getChecksumAddress(address: Hex) {
+  const unPrefixed = remove0x(address.toLowerCase());
+  const unPrefixedHash = remove0x(bytesToHex(keccak256(unPrefixed)));
+  return `0x${unPrefixed
+    .split('')
+    .map((character, nibbleIndex) =>
+      parseInt(unPrefixedHash[nibbleIndex] as string, 16) > 7
+        ? character.toUpperCase()
+        : character,
+    )
```

Comment:
Nit: to avoid the cast:

```suggestion
    .map((character, nibbleIndex) => {
      const hashCharacter = unPrefixedHash[nibbleIndex];
      if (!hashCharacter) {
        throw new Error('Hash shorter than address');
      }
      return parseInt(hashCharacter, 16) > 7
        ? character.toUpperCase()
        : character;
    })
```